sw/device/lib/dif/dif_otp_ctrl.h

```diff
@@ -0,0 +1,957 @@
+   */
+  kDifOtpCtrlStatusCodeCheckPending,
+} dif_otp_ctrl_status_code_t;
+
+enum {
+  /**
+   * The total number of `dif_otp_ctrl_status_code_t` values.
+   */
+  kDifOtpCtrlStatusCodesCount = kDifOtpCtrlStatusCodeCheckPending + 1,
+};
+
+/**
+ * A hardware-level error code, associated with a particular error defined in
+ * `dif_otp_ctrl_status_t`.
+ */
+typedef enum dif_otp_ctrl_error {
+  /**
+   * Indicates no error.
+   */
+  kDifOtpCtrlErrorOk,
+  /**
+   * Indicates an invalid command was issued by hardware.
+   *
+   * This error indicates non-recoverable hardware malfunction.
+   */
+  kDifOtpCtrlErrorInvalidHwCommand,
+  /**
+   * Indicates that the hardware initialization sequence failed.
+   *
+   * This error indicates non-recoverable hardware malfunction.
+   */
+  kDifOtpCtrlErrorFailedInit,
+  /**
+   * Indicates a recoverable error during a read operation.
+   *
+   * A followup read should work as expected.
+   */
+  kDifOtpCtrlErrorRecoverableRead,
+  /**
+   * Indicates an unrecoverable error during a read operation.
+   *
+   * This error indicates non-recoverable hardware malfunction.
+   */
+  kDifOtpCtrlErrorUnrecoverableRead,
+  /**
+   * Indicates an unspecified error during a read operation.
+   *
+   * This error indicates non-recoverable hardware malfunction.
+   */
+  kDifOtpCtrlErrorUnspecifiedRead,
+  /**
+   * Indicates that the blank write check failed during a write operation.
+   */
+  kDifOtpCtrlErrorBlankCheckFailed,
+  /**
+   * Indicates an unspecified error during a write operation.
+   *
+   * This error indicates non-recoverable hardware malfunction.
+   */
+  kDifOtpCtrlErrorUnspecifiedWrite,
+  /**
+   * Indicates an invalid command was issued to the DAI.
+   */
+  kDifOtpCtrlErrorInvalidDaiCommand,
+  /**
+   * Indicates a locked memory region was accessed.
+   */
+  kDifOtpCtrlErrorLockedAccess,
+  /**
+   * Indicates an integrity check failed in the buffer registers.
+   *
+   * This error indicates non-recoverable hardware malfunction.
+   */
+  kDifOtpCtrlErrorIntegrityLoss,
+  /**
+   * Indicates a consistency check failed in the buffer registers.
+   *
+   * This error indicates non-recoverable hardware malfunction.
+   */
+  kDifOtpCtrlErrorConsistencyLoss,
+  /**
+   * Indicates that the FSM of the controller is in a bad state.
+   *
+   * This error indicates that the device has been glitched by an attacker.
+   */
+  kDifOtpCtrlErrorFsmBadState,
+  /**
+   * Indicates that an escalation occured in the alert subsystem, and the
+   * controller's FSM is in its terminal state.
+   */
+  kDifOtpCtrlErrorFsmTerminalState,
+} dif_otp_ctrl_error_t;
+
+/**
+ * The overall status of the OTP controller.
+ *
+ * See `dif_otp_ctrl_get_status()`.
+ */
+typedef struct dif_otp_ctrl_status {
+  /**
+   * Currently active statuses, given as a bit vector. To check whether a
+   * particular status code was returned, write
+   *
+   *   bool has_code = (status.codes >> kMyStatusCode) & 1;
```

Comment:
This is a more general thing, but I'm not sure I'm actually a fan of the `kVariant = 1 << n;` pattern, since there's no way to really tell at the callsite that an enum is a bit mask; I think it's safer to just use bit indices and rely on `bitfield_bit32` functions.